mrp_production_request/models/mrp_production_request.py

```diff
@@ -0,0 +1,266 @@
+            self.env['mrp.production']._get_default_picking_type()),
+        required=True, readonly=True, states={'draft': [('readonly', False)]})
+    move_dest_ids = fields.One2many(
+        comodel_name='stock.move',
+        inverse_name='created_mrp_production_request_id',
+        string="Stock Movements of Produced Goods")
+    orderpoint_id = fields.Many2one(
+        comodel_name='stock.warehouse.orderpoint',
+        string='Orderpoint')
+
+    _sql_constraints = [
+        ('name_uniq', 'unique(name, company_id)',
+         'Reference must be unique per Company!'),
+    ]
+
+    @api.model
+    def _get_mo_valid_states(self):
+        return ['planned', 'confirmed', 'progress', 'done']
+
+    @api.multi
+    @api.depends('mrp_production_ids', 'mrp_production_ids.state', 'state')
+    def _compute_manufactured_qty(self):
+        valid_states = self._get_mo_valid_states()
+        for req in self:
+            done_mo = req.mrp_production_ids.filtered(
+                lambda mo: mo.state in 'done').mapped('product_qty')
+            req.done_qty = sum(done_mo)
+            valid_mo = req.mrp_production_ids.filtered(
+                lambda mo: mo.state in valid_states).mapped('product_qty')
+            req.manufactured_qty = sum(valid_mo)
+            req.pending_qty = max(req.product_qty - req.manufactured_qty, 0.0)
+
+    @api.multi
+    def _compute_mrp_production_count(self):
+        for rec in self:
+            rec.mrp_production_count = len(rec.mrp_production_ids)
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        self.product_uom_id = self.product_id.uom_id
+        self.bom_id = self.env['mrp.bom']._bom_find(
+            product=self.product_id, company_id=self.company_id.id,
+            picking_type=self.picking_type_id)
+
+    @api.multi
+    def _subscribe_assigned_user(self, vals):
+        self.ensure_one()
+        if vals.get('assigned_to'):
+            self.message_subscribe(
+                partner_ids=self.assigned_to.mapped('partner_id').ids)
+
+    @api.model
+    def _create_sequence(self, vals):
+        if not vals.get('name') or vals.get('name') == '/':
+            vals['name'] = self.env['ir.sequence'].next_by_code(
+                'mrp.production.request') or '/'
+        return vals
+
+    @api.model
+    def create(self, vals):
+        """Add sequence if name is not defined and subscribe to the thread
+        the user assigned to the request."""
+        vals = self._create_sequence(vals)
+        res = super(MrpProductionRequest, self).create(vals)
+        res._subscribe_assigned_user(vals)
+        return res
+
+    @api.multi
+    def write(self, vals):
+        res = super(MrpProductionRequest, self).write(vals)
+        for request in self:
+            request._subscribe_assigned_user(vals)
+        return res
+
+    @api.multi
+    def button_to_approve(self):
+        self.write({'state': 'to_approve'})
```

Comment:
this looks like old fashioned. IMHO it should be self.state='to_approve'

Reply:
this is not "old fashioned", it's the way to make it work with both a recordset or a single record.